contrib/hashicorp/consul/option.go

```diff
@@ -0,0 +1,60 @@
+package consul
+
+import "math"
+
+const (
+	serviceName = "consul"
+	spanName    = "consul.command"
+)
+
+type clientConfig struct {
+	serviceName   string
+	spanName      string
+	analyticsRate float64
+}
+
+// ClientOption represents an option that can be used to create or wrap a client.
+type ClientOption func(*clientConfig)
+
+func defaults(cfg *clientConfig) {
+	cfg.serviceName = serviceName
+	cfg.spanName = spanName
+	cfg.analyticsRate = math.NaN()
+}
+
+// WithServiceName sets the given service name for the client.
+func WithServiceName(name string) ClientOption {
+	return func(cfg *clientConfig) {
+		cfg.serviceName = name
+	}
+}
+
+// WithSpanName sets the given span name for the client.
```

Comment:
I think this is not needed. The span name should be the operation name: `LIST`, `KEYS`, `RELEASE`, `ACQUIRE`, etc. I know what you did is similar to `redis.command`, but I feel that's not the best way to approach this. Happy to discuss this and find the best way. You're going to use this so let me know what makes sense to you as a user.

Reply:
As discussed on Slack, let's keep this as `consul.command` throughout, but let's remove this option (`WithSpanName`). It's not something we used anywhere else.

Reply:
Removed `WithSpanName`